src/logical-container.ts

```diff
@@ -106,5 +106,8 @@ export const combineContainers = <T>(
       }
     }
   }
+  if (isObject(a) && "and" in a && Array.isArray(a.and) && a.and.length === 0)
+    return { and: [b as T] };
+
   return { and: [a as T, b as T] };
```

Comment:
I think I want to rewrite this in a slightly more generic way in order to cover same case (empty array) for `b`.

Reply:
done